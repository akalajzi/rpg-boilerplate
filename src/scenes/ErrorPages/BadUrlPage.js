// @flow strict
import React from 'react';
import styled from 'styled-components';

import defaultTheme from 'themes/default';

const FullPageContainer = styled.div`
  position: fixed;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0;
  padding: 0;
  width: 100%;
  height: 100%;
  background-color: ${({ theme }: ThemeProps) => theme.colors.black};
  color: ${({ theme }: ThemeProps) => theme.colors.white};
`;

FullPageContainer.defaultProps = {
  theme: defaultTheme,
};

const GiantText = styled.div`
  align-items: center;
  justify-content: center;
  font-size: 144px;
`;

const BadUrlPage = () => (
  <FullPageContainer>
    <GiantText>404</GiantText>
  </FullPageContainer>
);

export default BadUrlPage;
