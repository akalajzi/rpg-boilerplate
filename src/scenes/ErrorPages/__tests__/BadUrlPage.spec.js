// @flow strict

import * as React from 'react';
import renderer from 'react-test-renderer';

import BadUrlPage from '../BadUrlPage';

describe('#BadUrlPage', () => {
  test('render', () => {
    const component = renderer.create(<BadUrlPage />);

    expect(component).toMatchSnapshot();
  });
});
