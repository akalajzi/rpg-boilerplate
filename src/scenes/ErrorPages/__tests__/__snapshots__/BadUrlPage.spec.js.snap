// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#BadUrlPage render 1`] = `
.c0 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  padding: 0;
  width: 100%;
  height: 100%;
  background-color: #010101;
  color: #FFF;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 144px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    404
  </div>
</div>
`;
